File: DropDownList/MultiSelectDropDown/MultiSelectDropDownCSharp/CustomDropDownList.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing.Design;
using System.Linq;
using System.Text;
using Telerik.WinControls;
using Telerik.WinControls.UI;

namespace MultiSelectDropDown
{
    public class CustomDropDownList : RadDropDownList
    {
        public override string ThemeClassName
        {
            get
            {
                return typeof(RadDropDownList).FullName;
            }
            set
            {
            }
        }

        protected override RadDropDownListElement CreateDropDownListElement()
        {
            return new CustomEditorElement();
        }

        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content),
        Editor(typeof(CustomListControlCollectionEditor), typeof(UITypeEditor)),
        Category(RadDesignCategory.DataCategory)]
        [Description("Gets a collection representing the items contained in this RadDropDownList.")]
        public new RadListDataItemCollection Items
        {
            get
            {
                return base.Items;
            }
        }
    }

    public class CustomListControlCollectionEditor : Telerik.WinControls.UI.Design.RadListControlCollectionEditor
    {
        public CustomListControlCollectionEditor(Type itemType) : base(itemType)
        {
        }

        protected override Type[] CreateNewItemTypes()
        {
            Type[] baseTypes = base.CreateNewItemTypes();
            Type[] newTypes = new Type[baseTypes.Length + 1];
            baseTypes.CopyTo(newTypes, 0);
            newTypes[baseTypes.Length] = typeof(CustomListDataItem);
            return newTypes;
        }
   }


}

File: DropDownList/MultiSelectDropDown/MultiSelectDropDownVB/CustomDropDownList.vb
Imports Telerik.WinControls.UI
Imports System.ComponentModel
Imports System.Drawing.Design
Imports Telerik.WinControls

Public Class CustomDropDownList
    Inherits RadDropDownList
    Public Overrides Property ThemeClassName() As String
        Get
            Return GetType(RadDropDownList).FullName
        End Get
        Set(value As String)
        End Set
    End Property

    Protected Overrides Function CreateDropDownListElement() As RadDropDownListElement
        Return New CustomEditorElement()
    End Function

    <DesignerSerializationVisibility(DesignerSerializationVisibility.Content), Editor(GetType(CustomListControlCollectionEditor), GetType(UITypeEditor)), Category(RadDesignCategory.DataCategory)> _
    <Description("Gets a collection representing the items contained in this RadDropDownList.")> _
    Public Shadows ReadOnly Property Items() As RadListDataItemCollection
        Get
            Return MyBase.Items
        End Get
    End Property
End Class

Public Class CustomListControlCollectionEditor
    Inherits Telerik.WinControls.UI.Design.RadListControlCollectionEditor
    Public Sub New(itemType As Type)
        MyBase.New(itemType)
    End Sub

    Protected Overrides Function CreateNewItemTypes() As Type()
        Dim baseTypes As Type() = MyBase.CreateNewItemTypes()
        Dim newTypes As Type() = New Type(baseTypes.Length) {}
        baseTypes.CopyTo(newTypes, 0)
        newTypes(baseTypes.Length) = GetType(CustomListDataItem)
        Return newTypes
    End Function
End Class
